ledger/participant-integration-api/src/main/scala/platform/store/ActiveLedgerStateManager.scala

```diff
@@ -25,17 +25,40 @@ private[platform] class ActiveLedgerStateManager[ALS <: ActiveLedgerState[ALS]](
     initialState: => ALS
 ) {
 
+  // State that is restored on rollback.
+  private case class RollbackState(
+      createdIds: Set[ContractId],
+      archivedIds: Set[ContractId],
+      als: Option[ALS],
+  ) {
+    def mapAcs(f: ALS => ALS): RollbackState =
+      copy(als = als.map(f))
+  }
+
+  /** The accumulator state used while validating the transaction
+    * and to record the required changes to the ledger state.
+    * @param currentState The current state
+    * @param errs The list of errors produced during validation.
+    * @param parties The set of parties that are used
+    *  in the transaction and will be implicitly allocated
+    * @param archivedIds The set of contract ids
+    *  archived by the transaction or local contracts whose create has been rolled back.
+    */
   private case class AddTransactionState(
-      acc: Option[ALS],
+      currentState: RollbackState,
+      rollbackStates: List[RollbackState],
```

Comment:
What does the `rollbackStates` field represent? Is it a stack of rollback states? Might be worth documenting above.

Reply:
Added some comments in 33c5e4f9f808e5821bb34d3b469f03aba52b61af

Reply:
Thanks!